.github/workflows/integration_tests.yml

```diff
@@ -771,56 +625,132 @@ jobs:
           name: support-archive
           path: support-archive/*.zip
 
-  #######################################################################
-  # This job publishes a draft release based on the previous steps      #
-  #######################################################################
-  publish-draft-release:
-    name: Publish Draft release
+  publish-final-test-report:
+    name: Finalize tests and create reports
     needs: integration-test
     if: always() # always run, regardless of the outcome of the last job
     runs-on: ubuntu-20.04
     env:
       BRANCH: ${{ needs.integration-test.outputs.BRANCH }}
+      TEST_REPORTS_PATH: "./test-reports/"
+      FINAL_TEST_REPORTS_FOLDER: "./final-test-reports/"
+      FINAL_TEST_REPORT_FILEPATH_JSON: "./final-test-reports/final-test-report.json"
+      FINAL_TEST_REPORT_FILEPATH_MARKDOWN: "./final-test-reports/final-test-report.md"
     steps:
       - name: Check out code
         uses: actions/checkout@v2.4.0
 
-      - name: Debug - Output Branch
-        run: echo $BRANCH
+      - name: Set up Node
+        uses: actions/setup-node@v2.4.1
+        with:
+          node-version: 14
+      - run: npm install ndjson-parse@1.0.4
 
       - name: Download test reports
         uses: actions/download-artifact@v2
         with:
-          name: test-report
+          name: test-report-json
+          path: ${{ env.TEST_REPORTS_PATH }}
+
+      - name: Build multi-platform test report
+        uses: actions/github-script@v5.0.0
+        env:
+          TEST_REPORTS_PATH: ${{ env.TEST_REPORTS_PATH }}
+          FINAL_TEST_REPORTS_FOLDER: ${{ env.FINAL_TEST_REPORTS_FOLDER }}
+          FINAL_TEST_REPORT_FILEPATH_JSON: ${{ env.FINAL_TEST_REPORT_FILEPATH_JSON }}
+        with:
+          script: |
+            ndJsonParser = require('ndjson-parse');
+            fs = require('fs');
+            
+            const {TEST_REPORTS_PATH, FINAL_TEST_REPORT_FILEPATH_JSON, FINAL_TEST_REPORTS_FOLDER} = process.env
+            const jsonReportData = [];
+            
+            const fileList = fs.readdirSync(TEST_REPORTS_PATH);
+            fileList.forEach(fileName => {1
+                console.log(`Reading file: ${fileName}`);
+                const platformReportFile = fs.readFileSync(TEST_REPORTS_PATH + fileName, {encoding:'utf8', flag:'r'});
+                const platformTestReportRaw = ndJsonParser(platformReportFile);
+                
+                platformTestReportRaw.forEach(platformTestResult => {
+                    if(platformTestResult.test.includes("/")) {
+                        jsonReportData.push(platformTestResult);
+                    }
+                });
+            });
+            
+            const testResultMap = {};
+            
+            jsonReportData.forEach(testResult => {
```

Comment:
Group the test results by test name.
Result: map: key=testname, value=array of test results for the different platforms